dask/dataframe/core.py

```diff
@@ -2359,7 +2359,7 @@ def __getitem__(self, key):
         elif isinstance(key, slice):
             return self.loc[key]
 
-        if isinstance(key, list):
+        if isinstance(key, (list, pd.Index)):
```

Comment:
Here's what pandas allows: https://github.com/pandas-dev/pandas/blob/1c2844aca8e3dad7373576d5de40ee6810f7e5ad/pandas/core/frame.py#L2680

so we could either copy `if isinstance(key, (Series, np.ndarray, Index, list)):`, or ask pandas to make the allowed types public. I think just copying is fine.

Could you add pd.Series and ndarray, and add tests for those?

Reply:
Good point,  I'll add `pd.Series` and `np.ndarray` checks now. Thanks for linking to the corresponding pandas code!